### DISCUSSION TITLE
Is there a way to have a sort of schema.string.nullable() validation ?

### DISCUSSION THREAD

**Discussion by abdalem** [adonisjs/core#2113] [Help] [answered]
Hi ! I am using auth0 as an authentification 3rd party system, and adonis to manage the business logic of my app. I have built a logic that allow organizations to buy licences for their users, and these licence can be revoked to be available for another user. To perform this I instanciate a defined number of licences for each organization and set the userId (linked to auth0 userId) as null, so that means all available licences have userId as null. It is workin very well, except when i want to revoke a licence and set the userId as null again, to make this licence available again.

When I pass as body of my PATCH request `{ userId: null }`, my validator remove the key and I am not able to set it as null. My validation looks like this: 
```typescript
validateParams = async (request: RequestContract) => {
  const validationSchema = schema.create({
    userId: schema.string.optional(),
  })

  return await request.validate({
    schema: validationSchema,
  })
}
```

I would like to specify that for avoiding unwanted behavior, I would like revoke only if I specify `{ userId: null }`, so I need it to be null.

Thanks everyone by advance !!

**Comment by thetutlage** [comment, marked as answer]
This is planned and coming in the next release

**Reply by abdalem** [reply]
Is there any workaround currently ?

EDIT: I have found a workaround:
```typescript
  validateParams = async (request: RequestContract) => {
    const nullifiedUserId = request.only(['userId']);

    const validatedData = await request.validate({
      schema: schema.create({
        userId: schema.string.optional(),
      }),
    })

    return {
      ...nullifiedUserId,
      ...validatedData,
    }
  }
```

**Comment by yariksav** [comment]
Same question, Is there a way to have **ENUM** validation, like **schema.string.enum(['foo', 'bar'])** ?

**Reply by thetutlage** [reply]
@yariksav Read the docs, its already in there

**Reply by yariksav** [reply]
Thank you! 👍
